agent/lib/kontena/workers/service_pod_worker.rb

```diff
@@ -65,6 +92,16 @@ def on_container_event(topic, e)
       end
     end
 
+    # @param [String] topic
+    # @param [ContainerHealthCheckWorker::Event] event
+    def on_container_health_check(topic, event)
+      return if event.service_id != @service_pod.service_id
```

Comment:
Also `instance_number`

Reply:
This will also suffer from #2766, I imagine it could see "unhealthy" events for the old container after creating a new one.